### PR TITLE
categorize settings

### PR DESCRIPTION
**Pull request by Alizter**
We group settings into related categories.

### CODE REVIEW

File: editor/code/package.json [hunk 1]
```diff
@@ -173,5 +190,11 @@
     "typecheck": "tsc -b",
     "compile": "npm run esbuild",
     "watch": "npm run esbuild -- --watch"
+  },
+  "__metadata": {
```
Comment (ejgallego) [inline editor/code/package.json:194]: What is this?

File: editor/code/package.json [hunk 2]
```diff
@@ -143,11 +136,35 @@
               "Show on click and on cursor movement",
               "Show on click, cursor, and command movement"
             ]
+          }
+        }
+      },      {
+        "title": "Language Server",
```
Comment (ejgallego) [inline editor/code/package.json:142]: Maybe "Language Server Binary" ?

File: editor/code/package.json [hunk 3]
```diff
@@ -126,6 +108,17 @@
             "type": "boolean",
             "default": false,
             "description": "Show Coq's \"Notice\" messages as diagnostics, such as `About` and `Search` output."
+          }
+        }
+      },
+      {
+        "title": "Goals",
```
Comment (ejgallego) [inline editor/code/package.json:115]: Maybe "Info and Goals Panel"?

File: editor/code/package.json [hunk 4]
```diff
@@ -75,33 +75,20 @@
     ],
     "configuration": [
       {
+        "title": "General",
```
Comment (ejgallego) [inline editor/code/package.json:78]: Maybe "Proof Checking" or "Error Recovery" or maybe "Proof Checking and Error Recovery"?

**Review by ejgallego — COMMENTED** [review]
Nice! Some suggestions in comments.

File: editor/code/package.json (hunk 1 above)
Comment (Alizter) [reply editor/code/package.json:194]: VSCode generated it by accident, I'll remove it

**Comment by Alizter** [comment]
image

**Comment by Alizter** [comment]
@ejgallego I added a changelog.

**Comment by ejgallego** [comment]
How do I get to test this? I can't see where the hierarchy appears.

**Comment by Alizter** [comment]
@ejgallego In the settings panel under extensions. You might need to reload vscode at least twice.

**Review by ejgallego — APPROVED** [review]
Thanks a lot this is great!

File: editor/code/package.json [hunk 5]
```diff
@@ -173,5 +191,11 @@
     "typecheck": "tsc -b",
     "compile": "npm run esbuild",
     "watch": "npm run esbuild -- --watch"
+  },
+  "__metadata": {
```
Comment (ejgallego) [inline editor/code/package.json:195]: Umm, still this keeps appearing here, but it should not, right?

**Comment by ejgallego** [comment]
> @ejgallego In the settings window under extensions. You might need to reload vscode at least twice.

Ok got it, turns out you need to make the setting window large enough.

File: editor/code/package.json (hunk 5 above)
Comment (Alizter) [reply editor/code/package.json]: It's from `make ts-fmt`. It is quite annoying that compiling the project using `make extension` messes up all the formatting meaning I have to do `make ts-fmt` afterwards too, and that adds random crap here.

Comment (ejgallego) [reply editor/code/package.json]: I don't see any of that; but again you should never do `make extension` or `make ts-fmt` if you have vscode open.

I'd bet what you see here is a race between some watch target in vscode with some extensions you are using.

Comment (ejgallego) [reply editor/code/package.json]: BTW there is not way that make extension could modify any file in the source tree.

**Pull request merged**